tg/models/__init__.py

```diff
@@ -218,21 +218,20 @@ def add_message(self, chat_id, message):
         msg_id = message["id"]
         msg_set = self.msg_ids[chat_id]
         if msg_id in msg_set:
+            log.warning(
+                f"message {msg_id} was added earlier. probably, inaccurate "
+                "usage of the tdlib lead to unnecessary requests"
+            )
             return False
         log.info(f"adding {msg_id=} {message}")
         self.msgs[chat_id].append(message)
         msg_set.add(msg_id)
         return True
 
     def add_messages(self, chat_id, messages):
-        return any(self.add_message(chat_id, msg) for msg in messages)
-
-    def fetch_msgs(self, chat_id, offset=0, limit=10):
-        if offset + limit < len(self.msgs[chat_id]):
-            return sorted(self.msgs[chat_id], key=lambda d: d["id"])[::-1][
-                offset:limit
-            ]
+        return any([self.add_message(chat_id, msg) for msg in messages])
```

Comment:
when using `any()` with generator it behaves lazy and stop iteration on first non negative item. So, `add_message` was called only once each time :slightly_smiling_face: using list intead of generator helped